centreon/src/Core/Service/Infrastructure/Repository/DbReadServiceRepository.php

```diff
@@ -274,6 +274,39 @@ public function findServiceIdsLinkedToHostId(int $hostId): array
         return $serviceIds;
     }
 
+    /**
+     * @inheritDoc
+     */
+    public function findServiceIdsExclusivelyLinkedToHostId(int $hostId): array
+    {
+        $request = $this->translateDbName(
+            <<<'SQL'
+                SELECT hsr.service_service_id
```

Comment:
for a better performance i suggest this 
```suggestion
                SELECT hsr.service_service_id
FROM :db.host_service_relation hsr
INNER JOIN :db.host h
    ON h.host_id = hsr.host_host_id
INNER JOIN (
    SELECT service_service_id
    FROM :db.host_service_relation
    GROUP BY service_service_id
    HAVING COUNT(*) = 1
) uniq
    ON uniq.service_service_id = hsr.service_service_id
WHERE hsr.host_host_id = :host_id
AND h.host_register = '1';
```
wdyt ?

Reply:
great catch, it already looks better for 10 services : 
 before
<img width="256" height="41" alt="Image" src="https://github.com/user-attachments/assets/583d5069-64b4-4b0a-b376-b0efc82fede8" />
after

<img width="262" height="47" alt="Image" src="https://github.com/user-attachments/assets/79f44354-d18b-4aad-afe3-cca20f9f221a" />